### PR TITLE
fix: use the tag/commit as the aro version and copy image to the integration

### PR DESCRIPTION
**Pull request by petrkotas**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13493157
 
### What this PR does / why we need it:

ARO uses both tags and commits as its version.
The commits are used for the development scenario, tags are used when building and deploying to production.

Adds the pipeline step to copy the resulted image to the integration registry.

### Test plan for issue:
N/A

### Is there any documentation that needs to be updated for this PR?

Part of the pipeline documentation.

### CODE REVIEW

File: .pipelines/templates/template-push-images-to-acr-tagged.yml [hunk 1]
```diff
@@ -10,5 +10,17 @@ steps:
     az acr login --name "$RP_IMAGE_ACR"
     # azure checkouts commit, so removing master reference when publishing image
     export BRANCH=$(Build.SourceBranchName)
-    make publish-image-aro-tag
+    make publish-image-aro
```
Comment (petrkotas) [inline .pipelines/templates/template-push-images-to-acr-tagged.yml:13]: The change is in the makefile, which now identifies if the TAG was used, otherwise uses COMMIT

File: Makefile [hunk 2]
```diff
@@ -14,14 +13,22 @@ ifneq ($(shell uname -s),Darwin)
...
+endif
+
+ARO_IMAGE ?= $(ARO_IMAGE_BASE):$(VERSION)
+
 build-all:
 	go build -tags aro,containers_image_openpgp ./...
 
 aro: generate
-	go build -tags aro,containers_image_openpgp,codec.safe -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(COMMIT)" ./cmd/aro
+	go build -tags aro,containers_image_openpgp,codec.safe -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro
```
Comment (petrkotas) [inline Makefile:28]: I left the variable name to be `GitCommit` TAG and COMMIT has both point to the commit in the end. Also, changing the variable would make unnecessary changes to the code.

File: Makefile [hunk 3]
```diff
@@ -110,10 +105,10 @@ publish-image-proxy: image-proxy
 	docker push ${RP_IMAGE_ACR}.azurecr.io/proxy:latest
 
 proxy:
-	go build -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(COMMIT)" ./hack/proxy
+	go build -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./hack/proxy
 
 run-portal:
-	go run -tags aro -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(COMMIT)" ./cmd/aro portal
+	go run -tags aro -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro portal
```
Comment (m1kola) [inline Makefile:111]: Are we ok to have tag name as a `GitCommit` in go code?

cc @hawkowl for this. This might have an impact on what you are trying to do with ARO operator ACR env var [here](https://github.com/Azure/ARO-RP/pull/1845#discussion_r806681619).

**Review by s-amann — APPROVED** [review]

**Review by bryanro92 — COMMENTED** [review]
Thanks for these fixes. Overall lgtm, but will get some time Thursday or Friday to walk through this and the other documentation to validate myself.

File: Makefile (hunk 3 above)
Comment (hawkowl) [reply Makefile:111]: it appears like if there is no tag, the existing commit will be used instead. As long as it's consistently passed through, I don't think there'll be an effective change for development.

Comment (petrkotas) [reply Makefile:111]: this is exactly how it works! :)

**Review by m1kola — APPROVED** [review]

**Comment by m1kola** [comment]
@bryanro92 I'm going to merge as it is will be much easier to test pipelines. If you find any issues while looking into it - please ping @petrkotas about them.
